src/dmd/expressionsem.d

```diff
@@ -9240,13 +9240,16 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
             if (!verifyHookExist(exp.loc, *sc, Id._d_arraysetlengthTImpl, "resizing arrays"))
                 return setError();
 
+            auto lc = lastComma(exp.e2);
+            lc = lc.optimize(WANTvalue);
```

Comment:
Why is this necessary?

Reply:
1. To handle a chain `a.length = f.length = 0;` -> `(a = a[0..0] , (f = f[0..0] , 0));`
2. `optimize()` is needed for non-trivial cases:
```
    const x = 0;
    a.length = f.length = x;
```